syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package sqwak;

service Model_Manager {
  rpc GetFeatures (FeatureListRequest) returns (FeatureListResponse) {}
  rpc CreateModel (CreateModelRequest) returns (CreateModelResponse) {}
}

message FeatureListRequest {
  string name = 1;
}

message FeatureListResponse {
  string message = 1;
}

// CREATE MODEL ------------
message AudioSample {
  repeated int32 features = 2;
}

message MlClass {
  int32 label = 1;
  repeated AudioSample samples = 2;
}

message CreateModelRequest {
  string model_id = 1;
  repeated MlClass ml_classes = 2;
}

message CreateModelResponse {
  string message = 1;
}
